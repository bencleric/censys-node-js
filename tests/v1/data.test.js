const nock = require("nock");
const { ClientV1 } = require("../../src");
const { API_ID, API_SECRET } = require("../consts");

const seriesJson = {
  primary_series: [],
  raw_series: [],
};
const viewJson = {
  description:
    ("The Censys IPv4 dataset provides data about the services ",
    "(e.g., HTTP, SMTP, MySQL) running on all publicly-accessible IPv4 hosts."),
  results: { historical: [] },
};
const resultJson = { files: { filename: { file_type: "data" } } };
const testSeries = "ipv4_2018";
const testResult = "20200818";

describe("#censys.v1.data", () => {
  const c = new ClientV1(API_ID, API_SECRET).data;
  const baseUrl = c.baseUrl;
  const scope = nock(baseUrl);

  it("get series", async () => {
    scope.get("/data").reply(200, seriesJson);
    const res = await c.getSeries();
    expect(res).toStrictEqual(seriesJson);
  });

  it("view series", async () => {
    scope.get(`/data/${testSeries}`).reply(200, viewJson);
    const res = await c.viewSeries(testSeries);
    expect(res).toStrictEqual(viewJson);
  });

  it("view result", async () => {
    scope.get(`/data/${testSeries}/${testResult}`).reply(200, resultJson);
    const res = await c.viewResult(testSeries, testResult);
    expect(res).toStrictEqual(resultJson);
  });
});
