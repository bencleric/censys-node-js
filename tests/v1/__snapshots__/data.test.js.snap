// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#censys.v1.data get series 1`] = `
Object {
  "primary_series": Array [],
  "raw_series": Array [],
}
`;

exports[`#censys.v1.data view result 1`] = `
Object {
  "files": Object {
    "filename": Object {
      "file_type": "data",
    },
  },
}
`;

exports[`#censys.v1.data view series 1`] = `
Object {
  "description": "(e.g., HTTP, SMTP, MySQL) running on all publicly-accessible IPv4 hosts.",
  "results": Object {
    "historical": Array [],
  },
}
`;
